daffodil-tdml-processor/src/main/scala/org/apache/daffodil/tdml/processor/Runtime2TDMLDFDLProcessor.scala

```diff
@@ -117,8 +117,12 @@ final class Runtime2TDMLDFDLProcessorFactory private(
       val compileResult = if (generator.isError) {
         Left(generator.getDiagnostics) // C code compilation diagnostics
       } else {
-        // Create a processor for running the test using the executable
-        val processor = new Runtime2TDMLDFDLProcessor(tempDir, executable)
+        // Create a processor for running the test using the executable, passing it
+        // generator.diagnostics in order to let us check generator warnings later
+        val processor = new Runtime2TDMLDFDLProcessor(tempDir, executable, generator.getDiagnostics)
+        // Sadly, TDMLRunner never checks generator diagnostics in "Right" tuple below
+        // nor does it check processor diagnostics in cross tests (runtime2's TDML tests)
+        // unless you set defaultShouldDoWarningComparisonOnCrossTests true in RunnerFactory
```

Comment:
I wonder if there's a good solution for this? Perhaps we need a way to allow parser/unparseTestCases to enable this without checking the RunnerFactory? Maybe an option would to add an attribute tdml:errors/warnings that say which implementation that errors/warnings are for, e.g:
```scala
<tdml:errors implementation="daffodil">
  ...
</tdml:errors>
<tdml:errors implementation="daffodil-runtime2">
...
</tdml:errors>
```
Then errors can be checked specific to different implementations.

Not worth doing as part of this, but somethign we should maybe considered as an update to the TDML runner.